gemini/sample-apps/fashion-trends-prediction/app/pages/0_ Trend Prediction.py

```diff
@@ -0,0 +1,405 @@
+import base64
+import io
+import json
+from io import StringIO
+
+import streamlit as st
+import streamlit.components.v1 as components
+import vertexai
+import vertexai.preview.generative_models as generative_models
+from articles import Articles
+from config import config
+from genAIprompts import image_prompt, trends_prompt
+from prediction import Prediction
+from utilities import add_logo, stImg
+from utils_standalone_image_gen import predict_image
+from vertexai.preview.generative_models import GenerativeModel
+
+PROJECT_ID = config["PROJECT_ID"]  # @param {type:"string"}
+LOCATION = config["LOCATION"]  # @param {type:"string"}
+DATA_PATH = config["Data"]["current_data"]
+
+print("DATA_PATH: ", DATA_PATH)
+
+if "gemini_modal" not in st.session_state:
+    vertexai.init(project=PROJECT_ID, location=LOCATION)
+    st.session_state["gemini_modal"] = GenerativeModel("gemini-1.0-pro-vision-001")
+
+if "source" not in st.session_state:
+    st.session_state["source"] = "Insta"
+
+if "JSONdata" not in st.session_state:
+    with open(DATA_PATH, "r") as f:
+        st.session_state["JSONdata"] = json.load(f)
+
+    with open(DATA_PATH, "r") as f:
+        st.session_state["JSONdata2"] = json.load(f)
+
+
+def generate(image_gen_instruction):
```

Comment:
Are these wrapper functions really needed? All they do is call the gemini model.

Reply:
Removed them.